### PR TITLE
Refactorings

### PR DESCRIPTION
**Pull request by mykolamor**
This branch and feature request were meant for #103. But it turned out to be way too complex. Plus, the typescript grammar we use sucks and parses code incorrectly. Fixing it would be a lot of work. Thus, I have done all of the refactoring and preparations for this feature beforehand and put them in this request. Finally, I decided to temporarily ditch typescript and fall back to a custom [ECMAScript 2023](https://tc39.es/ecma262/2023/) grammar that I will write myself.

Reviewing commit by commit, you will see these changes:
1. a62f24df5f8047596df0ef400fe6e944926d2c5d Added the `access_modifier` node to the API, including C# `common` bindings.
2. 383425085e2589ea4cc988754e0cbc8530b82565 Rewrote typescript visitor behavior. Now, children to visit should be specified explicitly to skip parsing for nodes that are not yet implemented.
3. 607fd75644dc02598cd2d6ab63b6a29a198a347e Added an option to still parse all children, regardless if there is a handler for them.
4. aaa217b97816bc5a1f66a20d23c5dfbca485ac9b Added an ability to either `skip` not implemented nodes, `pass` through them without parsing, or `throw` an error to terminate the entire execution.
5. 0f8424c6fbd9924b7f38617451f024ea7c20eb47 Added empty files for the ECMAScript 2023 implementation.

Do note, however, that for ECMAScript 2023 our goal is to implement the syntax only, not the feature set and standard library. Even more, we don't have to implement the entire syntax now, but iteratively add features to it as we add new Crosslight nodes.

I'll merge after approval from the reviewers, but keep the branch for further work on the lexer and parser for ECMAScript 2023.

### CODE REVIEW

File: libs/c/core/include/core/nodes/access_modifier.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+#include <stdlib.h>
+#include "core/definitions.h"
+
+CL_BEGIN_C_DECLS
+
+/**
+ * @brief Access modifier type. Access modifiers do not support type combinations.
+ */
+enum cl_node_access_modifier_type {
+    access_modifier_none = 0, /**< No access modifier. Acts as a fallback, should not be used normally. */
```
Comment (SergiyIvan) [inline libs/c/core/include/core/nodes/access_modifier.h:22]: Just out of curiosity, what is this `/**< ... */` comment format?

File: libs/cpp/lang_ecmascript/private_include/lang_ecmascript/LexerEsBase.hpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    bool getStrictDefault();
+    void setUseStrictDefault(bool value);
+    bool IsStrictMode();
+    virtual std::unique_ptr<antlr4::Token> nextToken() override;
+    void ProcessOpenBrace();
+    void ProcessCloseBrace();
+    void ProcessStringLiteral();
+    bool IsRegexPossible();
+    void StartTemplateString();
```
Comment (SergiyIvan) [inline libs/cpp/lang_ecmascript/private_include/lang_ecmascript/LexerEsBase.hpp:30]: Is template string the same as interpolated string?
In what cases do we need to consider template depth?

**Review by SergiyIvan — APPROVED** [review]
Looks good, thanks!

File: libs/c/core/include/core/nodes/access_modifier.h (hunk 1 above)
Comment (mcmikecreations) [reply libs/c/core/include/core/nodes/access_modifier.h:22]: It's a postfix-style Doxygen documentation comment. I used these to not increase line count with prefix-style documentation comments on a previous line.

File: libs/cpp/lang_ecmascript/private_include/lang_ecmascript/LexerEsBase.hpp (hunk 2 above)
Comment (mcmikecreations) [reply libs/cpp/lang_ecmascript/private_include/lang_ecmascript/LexerEsBase.hpp:30]: It's the same. As for depth, can't say for sure right now, I'll bring it over to a separate issue when I get to implementing the lexer.

Comment (mcmikecreations) [reply libs/cpp/lang_ecmascript/private_include/lang_ecmascript/LexerEsBase.hpp:30]: Added #105.

**Pull request merged**